### PR TITLE
feat: changes to Dockerfile to publish to Dockerhub and associated RE…

### PR DESCRIPTION
**Pull request by sdewitt-newrelic**
## Updates
* Updated `Dockerfile` and `.dockerignore` to be _inclusive_ rather than _exclusive_
* Used `Dockerfile` to build and publish [image on Dockerhub](https://hub.docker.com/r/newrelic/newrelic-salesforce-exporter)
* Updated Docker documentation in`README.md` 
* Added `Dockerfile.sample` example for extending the base image
* Removed environments section from `newrelic.ini`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# NOTE: remove the "-slim" if you need a full featured base image.
-FROM python:3.11-slim
+FROM python:3.9
```
Comment (jbeveland27) [inline Dockerfile:1]: Is there a reason you went back to python 3.9?

File: README.md [hunk 2]
```diff
@@ -21,48 +21,225 @@ data to New Relic as either logs or events. The exporter currently supports
...
+environment variables, respectively. No command line argument are passed to the
+exporter.
 
-## Usage
+```bash
+docker run -t --rm --name salesforce-exporter \
+   -v "$PWD/config.yml":/usr/src/app/config.yml \
+   -e NR_LICENSE_KEY=$NR_LICENSE_KEY \
+   -e NEW_RELIC_APP_NAME="New Relic Salesforce Exporter" \
+   -e NEW_RELIC_LICENSE_KEY=$NEW_RELIC_LICENSE_KEY \
```
Comment (jbeveland27) [inline README.md:94]: Maybe I just haven't noticed it before, but having 2 `*_LICENSE_KEY` vars is confusing. Can we consolidate those?

File: .dockerignore [hunk 3]
```diff
@@ -1,14 +1,11 @@
-# Git stuff
-.git
-.github
-.gitignore
```
Comment (jbeveland27) [inline .dockerignore:4]: Just curious - is there a reason this was removed?

File: .dockerignore [hunk 4]
```diff
@@ -1,14 +1,11 @@
...
-
-# NR One Dashbords
-dashboards
-
-# Docs and samples
-*.md
-config.yml.sample
```
Comment (jbeveland27) [inline .dockerignore:11]: same question

File: Dockerfile (hunk 1 above)
Comment (sdewitt-newrelic) [reply Dockerfile:1]: Yes.  I can't recall now what it was but there is something in 3.9 that was removed in 3.11.

File: README.md (hunk 2 above)
Comment (sdewitt-newrelic) [reply README.md:94]: It _is_ confusing.  And irritating.  And I think you did notice it before.  We _can_ consolidate them but `NR_LICENSE_KEY` was used historically before we ever added the Python APM agent in so I left both for "backwards compatibility".  If we want to remove them moving forward, could you open an issue for it?

File: .dockerignore (hunk 3 above)
Comment (jbeveland27) [reply .dockerignore:4]: ah gotcha, I understand what you mean now w/ the inclusive comment

**Review by jbeveland27 — APPROVED** [review]

**Pull request merged**